### PR TITLE
Feature/Add holdings tab to account detail dialog (#2853)

### LINKED ISSUE #2853: Add holdings tab to account detail dialog
Add the _Holdings_ tab to the [AccountDetailDialog](https://github.com/ghostfolio/ghostfolio/tree/main/apps/client/src/app/components/account-detail-dialog) component.

account-detail-dialog-with-holdings

For implementation details, please check [HoldingsPageComponent](https://github.com/ghostfolio/ghostfolio/blob/main/apps/client/src/app/pages/portfolio/holdings/holdings-page.component.ts):
* Reuse [HoldingsTableComponent](https://github.com/ghostfolio/ghostfolio/tree/main/libs/ui/src/lib/holdings-table)
* Get data via `dataService.fetchPortfolioDetails()` with a filter on `accountId`

### PR DESCRIPTION
**Pull request by TobiasXy**
Closes #2853 

I recently started to get the urge to maybe contribute to some open source project and since I also started using Ghostfolio on my server I picked the first issue and gave it a try. Please let me know if this PR looks acceptable :D

### CODE REVIEW

File: apps/client/src/app/components/account-detail-dialog/account-detail-dialog.html [hunk 1]
```diff
@@ -69,6 +69,14 @@
       [mat-stretch-tabs]="false"
       [ngClass]="{ 'd-none': isLoadingActivities }"
     >
+      <mat-tab>
+        <ng-template i18n mat-tab-label>Holdings</ng-template>
+        <gf-holdings-table
+          [baseCurrency]="user?.settings?.baseCurrency"
+          [locale]="user?.settings?.locale"
```
Comment (dtslvr) [inline apps/client/src/app/components/account-detail-dialog/account-detail-dialog.html:76]: Please add `[deviceType]="deviceType"`

File: apps/client/src/app/components/account-detail-dialog/account-detail-dialog.module.ts [hunk 2]
```diff
@@ -28,7 +29,8 @@ import { AccountDetailDialog } from './account-detail-dialog.component';
     MatButtonModule,
     MatDialogModule,
     MatTabsModule,
-    NgxSkeletonLoaderModule
+    NgxSkeletonLoaderModule,
+    GfHoldingsTableModule
```
Comment (dtslvr) [inline apps/client/src/app/components/account-detail-dialog/account-detail-dialog.module.ts:33]: Please move up to keep the alphabetical order

File: apps/client/src/app/components/account-detail-dialog/account-detail-dialog.component.ts [hunk 3]
```diff
@@ -83,6 +86,26 @@ export class AccountDetailDialog implements OnDestroy, OnInit {
...
+        filters: [
+          {
+            type: 'ACCOUNT',
+            id: this.data.accountId
+          }
+        ]
+      })
+      .pipe(
+        map((d) =>
+          Object.values(d.holdings).map(function (x) {
```
Comment (dtslvr) [inline apps/client/src/app/components/account-detail-dialog/account-detail-dialog.component.ts:100]: Can you move that transformation directly to `fetchPortfolioDetails()`?

You can then even remove the same code in `HoldingsPageComponent` 🙂

File: apps/client/src/app/components/account-detail-dialog/account-detail-dialog.component.ts [hunk 4]
```diff
@@ -55,6 +57,7 @@ export class AccountDetailDialog implements OnDestroy, OnInit {
   public transactionCount: number;
   public user: User;
   public valueInBaseCurrency: number;
+  public holdings$: Observable<PortfolioPosition[]>;
```
Comment (dtslvr) [inline apps/client/src/app/components/account-detail-dialog/account-detail-dialog.component.ts:60]: Can you please move up to keep the alphabetical order?

**Review by dtslvr — CHANGES_REQUESTED** [review]
Thanks a lot for your contribution, @TobiasXy! 👍🏻
I have added a few comments.

Ps. Next time please leave a comment in the issue, that you are about to work on it. This prevents possible redundant effort.

File: apps/client/src/app/components/account-detail-dialog/account-detail-dialog.html (hunk 1 above)
Comment (TobiasXy) [reply apps/client/src/app/components/account-detail-dialog/account-detail-dialog.html:76]: Sure, but do you mind telling me what this field is about though? It doesn't seem to be used and my IDE is even complaining about an "unresolved variable".

Comment (TobiasXy) [reply apps/client/src/app/components/account-detail-dialog/account-detail-dialog.html:76]: Ok, found it. It's actually data.deviceType here, that's why it said "unresolved"

**Comment by TobiasXy** [comment]
And sorry, next time I will definitely leave a comment first 😅

File: apps/client/src/app/components/account-detail-dialog/account-detail-dialog.html (hunk 1 above)
Comment (dtslvr) [reply apps/client/src/app/components/account-detail-dialog/account-detail-dialog.html:77]: > Sure, but do you mind telling me what this field is about though?

The visible columns are controlled via `deviceType`. On mobile, the holdings table is simplified.

> It's actually data.deviceType here

Oops! 🙈

**Review by dtslvr — APPROVED** [review]
Thank you for the adaptions. Looks fine now 👍🏻 Ready to merge.

Let me know if you are interested in solving more issues 🙂

**Comment by TobiasXy** [comment]
Sure. I might not always have time, but I guess working on some issue here or there would be nice :)

**Pull request merged**